### ISSUE TITLE
v1.0.0 Release QA

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#154]
v1.0.0 Release Team: @colin-axner @AdityaSripal 

Once the 1.0.0 milestone is complete, we will create a `release/v1.0.0` branch which will be used to cut a `v1.0.0beta1` tag. State breaking or relayer breaking changes will not be backported to this branch unless necessary to fix bugs. Note, we do reserve the right to introduce state breaking or API breaking changes during beta in cases where the change would provide a substantial benefit to the codebase. This will be clearly noted in the changelog. Once the beta release has gone through the below testing, we will do a release candidate and do a final round of testing/poshing. If all goes well, the final release will be made following the release candidate.  

**Beta Requirements:**
- [x] Frozen Height pr #165  (ready for review)
- [x] Block delay parameter #100 
- [x] Update to SDK beta tag #167 (draft)

**Audit Fixes needed:**
- [x] #181 (r4r)
- [x] #184 (r4r)
- [x] #185
- [x] #191

**Nice to haves** (only included if merged before the above requirements):
- [x] Remove misbehaviour GetHeight #166
- [x] Change solomachine frozen sequence to boolean #169 

**RC Requirements:**
- [x] High level internal audit of changes in the CHANGELOG (audit finished, minor fixes needed)

  **Lengthy Reviews (45-60 min each)**
  - [x] UpdateClient Proposal (audit done, needs fixes #176)
  - [x] IBC Upgrade Proposal
  - [x] Time Monotonicity pr
  - [x] Consensus state key efficient iteration + pruning (metadata key setting/deleting needed)
  
  **Quick Reviews: 10 min each**
  - [x] OnRecvPacket callback support for revert state
  - [x] IsFrozen -> Status (#186 - nice-to-have)
  - [x] Solo Machine Fixes
  - [x] Max TM chain-id length
  
- [x] Upgrade Testing
  - [x] Upgrading from SDK v0.42.x -> v1.0.0 ibc-go with x/upgrade prop
  - [x] Upgrading from SDK v0.42.x -> v1.0.0 ibc-go with x/upgrade prop and genesis restart
  - [x] Upgrading from ibc-go v1.0.0 -> v1.0.0 ibc-go with ibc upgrade prop
  - [x] Upgrading from ibc-go v1.0.0 -> v1.0.0 ibc-go with ibc upgrade prop and genesis restart
- [x] Cross version testing
  - [x] Test using Hermes relaying between a chain on SDK v0.42.x and a chain using ibc-go v1.0.0
  - [x] Test using ts-relayer relaying between a chain on SDK v0.42.x and a chain using ibc-go v1.0.0
- [x] Migration scripts (deleting solo-machines, removing expired consensus states) #11 

Once the above is checked off, we will do a final release after the SDK finalizes their release of v0.43.0
  
____

#### For Admin Use

- [x] Not duplicate issue
- [x] Appropriate labels applied
- [x] Appropriate contributors tagged/assigned

**Comment by colin-axner** [comment]
Here is an update on this release cycle:

Currently we are in alpha. We have completed the scope of the features for this release. We have completed compatibility testing with Hermes and the ts-relayer. We have done an internal audit of the change log and implemented fixes for almost all of the bugs found

We believe it is important to make the currently scoped changes available as soon as possible.  Since we are very close to finalizing a release, we are continuing with the original scope of this v1.0.0 release. There are two issues left that will be included in this release. They are #11 and #191, both of which relate to correctly handling genesis exporting and x/upgrade migrations. After these issues are fixed, we will create a release branch (`release/v1.0.0`) and do a `beta1` release. We made create the release branch early so as not to block development of features not scoped for this release. The testing left to do is for upgrading from SDK v0.42.x to ibc-go v1.0.0 and doing an IBC client breaking upgrade using the upgrade logic included in the proposed v1.0.0. Following this testing, we will be ready to do `rc`'s. We aim to finalize a release as soon as the SDK v0.43.0 release is completed

There is exciting work around relayer incentivization and we want to prioritize making that work available as soon as it is implemented and gone through sufficient QA. For this reason, we reserve the right to include these changes in v1.0.0 if they are implemented and tested (with relayer support) before the SDK does a v0.43.0 release. If a v1.0.0 is released without relayer incentivization, we will do a ibc-go v2.0.0 as soon as relayer incentivization is implemented has gone through sufficient QA

**Comment by colin-axner** [comment]
`beta1` has been released. I'm out on vacation for 2 weeks. What's left to do before a final release is final testing of the last 3 upgrade scenarios (I have done partial testing)

**Comment by colin-axner** [comment]
`rc0` is out. We are done testing and have no blocking issues. To further track the `v1.0.0` final release you can follow the 1.0.0 milestone. As of right now, we are simply waiting for a final `v0.43.0` SDK release and then would follow with a final `v1.0.0` ibc-go release

The main branch will move forward with `v2.0.0` development. You can follow the issues for that on the `2.0.0` milestone
